WalletWasabi/CoinJoin/Client/Clients/CoinJoinClient.cs

```diff
@@ -445,7 +445,9 @@ private async Task RegisterOutputAsync(ClientRound ongoingRound)
 						else
 						{
 							// Should never happen, but oh well we can autocorrect it so why not.
-							ExposedLinks[input] = ExposedLinks[input].Append(new HdPubKeyBlindedPair(DestinationKeyManager.GetKeyForScriptPubKey(activeOutput.Address.ScriptPubKey) ?? KeyManager.GetKeyForScriptPubKey(activeOutput.Address.ScriptPubKey), false));
+							DestinationKeyManager.TryGetKeyForScriptPubKey(activeOutput.Address.ScriptPubKey, out HdPubKey? hdPubKey);
+							KeyManager.TryGetKeyForScriptPubKey(activeOutput.Address.ScriptPubKey, out HdPubKey? hdPubK);
+							ExposedLinks[input] = ExposedLinks[input].Append(new HdPubKeyBlindedPair(hdPubKey ?? hdPubK, false));
```

Comment:
@nopara73 Is there a better way to write this?

Reply:
I don't have the right to commit, so i will just copy the souliton here

Reply:
``` else
						{
							// Should never happen, but oh well we can autocorrect it so why not.
							if (!DestinationKeyManager.TryGetKeyForScriptPubKey(activeOutput.Address.ScriptPubKey, out HdPubKey? hdPubKey)
								&& !KeyManager.TryGetKeyForScriptPubKey(activeOutput.Address.ScriptPubKey, out hdPubKey))
							{
								throw new NotSupportedException("Couldn't get script.");
							}
							ExposedLinks[input] = ExposedLinks[input].Append(new HdPubKeyBlindedPair(hdPubKey, false));
						}```

Reply:
@adamPetho I edited the code, thank you.